plugins/toolbox-search/src/block_searcher.ts

```diff
@@ -29,12 +29,30 @@ export class BlockSearcher {
       this.indexBlockText(blockType.replaceAll('_', ' '), blockType);
       block.inputList.forEach((input) => {
         input.fieldRow.forEach((field) => {
+          this.indexDropdownOption(field, blockType);
           this.indexBlockText(field.getText(), blockType);
         });
       });
     });
   }
 
+  /**
+   * Check if the field is a dropdown, and index every text in the option
+   * @param field We need to check the type of field
+   * @param blockType The block type to associate the trigrams with.
+   */
+  private indexDropdownOption(field: Blockly.Field, blockType: string) {
+    if (field instanceof Blockly.FieldDropdown) {
+      field.getOptions(true).forEach((option) => {
+        if (typeof option[0] === 'string') {
+          return this.indexBlockText(option[0], blockType);
+        } else if ('alt' in option[0]) {
+          return this.indexBlockText(option[0].alt, blockType);
```

Comment:
I am not sure if this would call every option in dropdown field.
Maybe : `this.indexBlockText(option[0].alt, blockType);`
but I will leave to @johnnesky to decide.

Reply:
We are calling `option[0]` because the MenuOption type is an array containing 2 elements, and we only care about the first element, which is the human-readable value (text or image) (defined here: https://github.com/google/blockly/blob/e4d8f160aa72649220aebc25e4eef6bd5ea267bf/core/field_dropdown.ts#L674)

The if / else statement here is handling cases when `option[0]` is a `string`, or a `ImageProperties`.

The actual logic looping through every option in the dropdown field is on line 46: `field.getOptions(true).forEach((option) => {...})`

Reply:
Good catch! You are right, we shouldn't be `return`-ing anything, and we just need to call `this.indexBlockText()` to generate trigrams. 

@sammitako can you remove the word `return` from line 48 and line 50? The rest should keep the same and do the work.

Reply:
Thank you for the catch @humera811. 
@ludizhan I just removed `return` on line 48 and line 50.